.circleci/config.yml

```diff
@@ -1,38 +1,29 @@
-version: 2
+version: 2.1
 jobs:
     build:
         docker:
-            - image: circleci/python:3.7.3-stretch
+            - image: cimg/python:3.9
         steps:
             - checkout
             - restore_cache:
                 keys:
-                  - data-cache-{{ checksum "./autoreject/__init__.py" }}  # restore cache from last build. Unless __init__.py has changed since then
+                  # restore cache from last build. Unless __init__.py has changed since then
+                  - data-cache-{{ checksum "./autoreject/__init__.py" }}
             - run:
-                name: Install miniconda
+                name: Setup Python environment
                 command: |
-                  wget -q https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh;
-                  chmod +x ~/miniconda.sh;
-                  ~/miniconda.sh -b -p ~/miniconda;
-                  echo "export PATH=~/miniconda/bin:$PATH" >> $BASH_ENV;
+                    python -m pip install --user --upgrade --progress-bar off pip
+                    python -m pip install --user --progress-bar off -e .[full]
+                    python -m pip install --user --progress-bar off -e .[doc]
             - run:
-                name: Setup Python environment
+                name: Download MNE sample data
                 command: |
-                    conda update --yes --quiet conda
-                    conda create -n testenv --yes python=3.7
-                    source activate testenv
-                    conda install --yes --quiet numpy matplotlib scipy scikit-learn joblib
-                    conda install --yes --quiet cython pillow
-                    pip install --upgrade pip
-                    pip install tqdm
-                    pip install https://api.github.com/repos/mne-tools/mne-python/zipball/master
-                    pip install sphinx sphinx-gallery sphinx_bootstrap_theme numpydoc
-                    pip install -e .
+                  python -c "import mne; mne.datasets.sample.data_path()"
             - run:
                 name: Build the documentation
                 command: |
-                    source activate testenv
-                    cd doc && make html
+                    make -C doc html
+                no_output_timeout: 4h
```

Comment:
4 hours without output??

Reply:
yes, that was before I figured out that the `-v` is an important `SPHINXOPT` so that circleci doesn't stop running. I'll reset this to 2 ... and we can further reduce once we speed up the examples.